Junimatic/Patcher.cs

```diff
@@ -0,0 +1,158 @@
+
+                    __result = true; // Set the result returned to the code calling addItemToInventoryBool to true to indicate item successfully added
+                    return false; // Return false to suppress addItemToInventoryBool from running
+                }
+            }
+            catch (Exception ex)
+            {
+                ModEntry.Instance.LogError($"Failed in {nameof(Farmer_addItemToInventoryBool_Prefix)}:\n{ex}");
+            }
+
+            return true; // Return true to run addItemToInventoryBool normally
+        }
+
+        /// <summary>
+        /// Prefix patch to <see cref="Game1.createItemDebris"/>.
+        /// When called during Junimatic harvesting logic, collect the item into <see cref="collectedItems"/>
+        /// instead of the usual implementation.
+        /// </summary>
+        private static bool Game1_createItemDebris_Prefix(ref Debris __result, Item item)
+        {
+            try
+            {
+                if (collectedItems is not null) // If inside InterceptHarvest
+                {
+                    AddToItemList(collectedItems, item);
+                    __result = null!; // Set the result returned to the code calling createItemDebris to null as no Debris is created (returned result is not used)
+                    return false; // Return false to suppress createItemDebris from running
+                }
+            }
+            catch (Exception ex)
+            {
+                ModEntry.Instance.LogError($"Failed in {nameof(Game1_createItemDebris_Prefix)}:\n{ex}");
+            }
+
+            return true; // Let the method run as normal
+        }
+
+        /// <summary>
+        /// Prefix patch to <see cref="Game1.createObjectDebris"/>.
+        /// When called during Junimatic harvesting logic, collect the item into <see cref="collectedItems"/>
+        /// instead of the usual implementation.
+        /// </summary>
+        private static bool Game1_createObjectDebris_Prefix(string id)
+        {
+            try
+            {
+                if (collectedItems is not null) // If inside InterceptHarvest
+                {
+                    AddToItemList(collectedItems, ItemRegistry.Create(id));
+                    return false; // Return false to suppress createObjectDebris from running
+                }
+            }
+            catch (Exception ex)
+            {
+                ModEntry.Instance.LogError($"Failed in {nameof(Game1_createObjectDebris_Prefix)}:\n{ex}");
+            }
+
+            return true; // Let the method run as normal
+        }
+
+        /// <summary>
+        ///   Adds <paramref name="item"/> to <paramref name="list"/> while merging stacks of common items.
+        /// </summary>
+        private static void AddToItemList(List<Item> list, Item item)
+        {
+            var extantItem = list.FirstOrDefault(x => x.ItemId == item.ItemId && x.Quality == item.Quality && x.Stack + item.Stack < 1000);
```

Comment:
Not worth handling the possibility of multiple ColoredObjects? I'm not aware of any "machines" that output multiple ColoredObjects of varying tint, but I could imagine so scenarios: a bush that produces multiple colors of flowers in one harvest, a super/split preserves jar that can handle multiple produce at one time, etc.

Reply:
Again, very likely I'm doing it wrong.  I don't understand how ColoredObjects work at all.

Reply:
Functionally, ColoredObjects are the same as Objects. The only difference is the additional field, color, and the draw functions. Even the constructor calls the base Object constructor, only taking an additional parameter: color. There's not much else to know about ColoredObjects. They stack on the same criteria as Objects, with the addition of color.

So when searching for an existing stack, in addition to checking the ItemId and Quality (and count <1000), also check on color. I say that like it's simple to implement, but its not quite straightforward since color is specifically a ColoredObject field and we're working with Items in this method. The Items need to be cast as ColoredObjects (if they are) before they can be compared: if (Item is ColoredObject ColoredObject) ColoredObject.color.Value. Could warrant it's own bool method taking the current list item and item to check for.

Reply:
I extended the comparison logic to include color here and where it decides whether or not there's room in chests.  That code could sore use some unit tests.  Also, it'd be nice if the code that draws the items being carried by Junimos was colored-object-aware too.